components/brave_shields/core/browser/ad_block_default_resource_provider.cc

```diff
@@ -37,48 +36,43 @@ AdBlockDefaultResourceProvider::AdBlockDefaultResourceProvider(
 AdBlockDefaultResourceProvider::~AdBlockDefaultResourceProvider() = default;
 
 base::FilePath AdBlockDefaultResourceProvider::GetResourcesPath() {
-  if (component_path_.empty()) {
-    // Since we know it's empty return it as is.
-    return component_path_;
+  if (!component_reader_) {
+    return base::FilePath();
   }
 
-  return component_path_.AppendASCII(kAdBlockResourcesFilename);
+  return component_reader_->GetComponentRootDeprecated().AppendASCII(
+      kAdBlockResourcesFilename);
 }
 
 void AdBlockDefaultResourceProvider::OnComponentReady(
-    const base::FilePath& path) {
-  component_path_ = path;
-  base::FilePath resources_path = GetResourcesPath();
-
-  if (resources_path.empty()) {
-    // This should not happen, but if it does, we should not proceed.
-    return;
-  }
+    std::unique_ptr<component_updater::ComponentContentsReader> reader) {
+  component_reader_ = std::move(reader);
 
   // Load the resources (as a string)
-  base::ThreadPool::PostTaskAndReplyWithResult(
-      FROM_HERE, {base::MayBlock()},
-      base::BindOnce(&brave_component_updater::GetDATFileAsString,
-                     resources_path),
+  LoadResources(
       base::BindOnce(&AdBlockDefaultResourceProvider::NotifyResourcesLoaded,
                      weak_factory_.GetWeakPtr()));
 }
 
 void AdBlockDefaultResourceProvider::LoadResources(
     base::OnceCallback<void(const std::string& resources_json)> cb) {
-  base::FilePath resources_path = GetResourcesPath();
-  if (resources_path.empty()) {
-    // If the path is not ready yet, run the callback with empty resources to
-    // avoid blocking filter data loads.
+  if (!component_reader_) {
+    // If the component is not ready yet, run the callback with empty resources
+    // to avoid blocking filter data loads.
     std::move(cb).Run("[]");
     return;
   }
 
-  base::ThreadPool::PostTaskAndReplyWithResult(
-      FROM_HERE, {base::MayBlock()},
```

Comment:
the default priority is `USER_BLOCKING`, but we changed it to `BEST_EFFORT`
https://source.chromium.org/chromium/chromium/src/+/main:base/task/task_traits.h;l=239?q=base::MayBlock

We need to return the original priority.